hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncTableImpl.java

```diff
@@ -497,7 +497,7 @@ public void run(MultiResponse resp) {
                     "Failed to mutate row: " + Bytes.toStringBinary(mutation.getRow()), ex));
               } else {
                 future.complete(respConverter
-                  .apply((Result) multiResp.getResults().get(regionName).result.get(0)));
+                  .apply((RES) multiResp.getResults().get(regionName).result.get(0)));
```

Comment:
OK, so the problem here is that, the result for a multi operation is an Object? And we will have two types, one is Result, for normal mutateRow, and the other is CheckAndMutateResult, for checkAndMutate?

Reply:
Yes, correct. Previously, we returned only Result type but after this fix, we will return Result (for normal mutateRow) and CheckAndMutateResult (for checkAndMutate). So I used a generic type here.

Reply:
Is there any problem here?